src/mlpack/core/cv/metrics/top_k_accuracy_impl.hpp

```diff
@@ -0,0 +1,61 @@
+/**
+ * @file top_k_accuracy_impl.hpp
+ * @author Arunav Shandeelya
+ * 
+ * The implementation of class Top_K_Accuracy_score
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_CORE_CV_METRICS_TOPK_ACCURACY_IMPL_HPP
+#define MLPACK_CORE_CV_METRICS_TOPK_ACCURACY_IMPL_HPP
+
+namespace mlpack {
+namespace cv {
+
+template<typename MLAlgorithm, typename DataType, typename TopK>
+double TopKAccuracy::Evaluate(MLAlgorithm& model,
+                        const DataType& data,
+                        const arma::Row<size_t> labels,
+                        const DataType& predsprob,
+                        const TopK& k)
+{
+  if (data.n_cols != labels.n_cols)
+  {
+    std::ostringstream fpp;
+    fpp << "TopK_Accuracy::Evaluate(): number of points (" << data.n_cols << ")"
+        << "doesn't matches number of labels (" << labels.n_cols << ")!"
+        << std::endl;
+    throw std::invalid_argument(fpp.str());
+  }
+  // Taking Classification output from the model.
+  model.Classify(data, predsprob);
+  // Shape of the predicted probabilities.
+  double predictedProb = predsprob.size();
+  // Top 'k' label prediction probabilities.
+  if (k < predsprob[1])
+  {
+    size_t idx = predictedProb[1] - k - 1;
+  else
+  {
+    std::ostringstream fs;
+    fs << "Less number of class for Top K Accuracy score" << std::endl;
+  }
+  float count = 0;
+  double srt = arma::sort_index(predictionProbs);
+  for (size_t i = 1; i < predictedProb[0]; i++)
+  {
+    if (labels[i] != srt[i, idx + 1:])
+    {
+      count = count + 1;
+    }
```

Comment:
There are 3 things that would probably need fixing here.

1. There is predictionProbs declared before this, may be it refers to predsProb.
2. sort_index returns vector of sorted indices, which can' t be stored in double.
3. I am not sure what `[i, idx + 1:]` refers to. `:` syntax is for python and I am not sure if armadillo has any such functionality. To span either use `arma::span(...)` or appropriate function from armadillo. You can refer to armadillo's [documentation](http://arma.sourceforge.net/docs.html#sort_index) for more info.

Hopefully, These comments are helpful in fixing the build. Let me know if you need any help or clarification.

Regards,
Thanks.

Reply:
@kartikdutt18 Ohh, I am so sorry. Actually, I have more practice in python than C++. I will do the changes accordingly. Thanks for addressing.

Reply:
No worries at all, Happy to help out in anyway. Thanks.